plugins/vue-directives.js

```diff
@@ -0,0 +1,36 @@
+import Vue from 'vue';
+
+Vue.directive('visible-on-scroll', {
+  inserted: (el) => {
+    el.scrolledVisible = true;
+    window.addEventListener('scroll', () => handleScroll(el));
```

Comment:
I can assume there are no issues with window being undefined due to SSR?

Reply:
haven't seen any ssr issues while testing, but just to be safe I've added this conditional so it'll only run client-side

   ```
if (process.browser) {
      window.addEventListener('scroll', () => handleScroll(el));
      window.addEventListener('orientationchange', () => handleOrientationChange(el));
}
```